pkg/yttlibrary/regexp.go

```diff
@@ -47,3 +48,67 @@ func (b regexpModule) Match(thread *starlark.Thread, f *starlark.Builtin, args s
 
 	return starlark.Bool(matched), nil
 }
+
+func (b regexpModule) Replace(thread *starlark.Thread, f *starlark.Builtin, args starlark.Tuple, kwargs []starlark.Tuple) (starlark.Value, error) {
+	if args.Len() != 3 {
+		return starlark.None, fmt.Errorf("expected exactly 3 arguments")
+	}
+
+	pattern, err := core.NewStarlarkValue(args.Index(0)).AsString()
+	if err != nil {
+		return starlark.None, err
+	}
+
+	re, err := regexp.Compile(pattern)
+	if err != nil {
+		return starlark.None, err
+	}
+
+	source, err := core.NewStarlarkValue(args.Index(1)).AsString()
+	if err != nil {
+		return starlark.None, err
+	}
+
+	repl := args.Index(2)
+	switch repl.(type) {
+	case *starlark.Function:
+		return b.replaceLambda(thread, re, source, repl.(*starlark.Function))
+	default:
+		return b.replaceString(re, source, repl)
+	}
+}
+
+func (b regexpModule) replaceString(re *regexp.Regexp, source string, repl starlark.Value) (starlark.Value, error) {
+	replStr, err := core.NewStarlarkValue(repl).AsString()
+	if err != nil {
+		return starlark.None, err
+	}
+
+	newString := re.ReplaceAllString(source, replStr)
+
+	return starlark.String(newString), nil
+}
+
+func (b regexpModule) replaceLambda(thread *starlark.Thread, re *regexp.Regexp, source string, repl *starlark.Function) (result starlark.Value, err error) {
+	defer func() {
+		if r, ok := recover().(error); ok {
+			err = r
+		}
+	}()
+
+	newString := re.ReplaceAllStringFunc(source, func(match string) string {
+		args := starlark.Tuple{starlark.String(match)}
+		result, err := starlark.Call(thread, repl, args, []starlark.Tuple{})
+		if err != nil {
+			panic(err)
+		}
+		newString, err := core.NewStarlarkValue(result).AsString()
```

Comment:
I think this is the correct way to extract the Go string value from the starlark.Value result? We can't simply do `result.String()`, as that adds quotes to the string representation.